### DISCUSSION TITLE
Port Forwarding Issues

### DISCUSSION THREAD

**Discussion by peerkleio** [community/community#28252] [Codespaces]
We have multiple separate systems using Codespaces and in the last hour or so we have been running into issues related to port forwarding. Specifically we are seeing this error,

`Basis tunnel and tunnel ports must be initialized`

In VSCode when initializing a codespace and when using the `gh cs ssh` command.  Below are entries we see in the exported codespace logs,

```
[2022-08-03 20:07:42.644 ContainerTunnel I] ContainerTunnelManager:ProcessConnectionsAsync - Waiting for connections on port '3002'
[2022-08-03 20:07:42.893 ContainerTunnel I] process exited with exit code 1

[2022-08-03 19:27:50.531 Client:extension:3714:resolver:onSharingSucceeded:3000:0.0.0.0:start I] opening the tunnel> object #0: {
  "host": "0.0.0.0",
  "port": 3000
}
[2022-08-03 19:27:50.535 Client:extension:3714:resolver:onSharingSucceeded:3000:0.0.0.0:start E] failed to open tunnel> object #0: cannot open tunnel
  at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:106:49459
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at handlePortSharingSucceededEvent (/Users/peerallan/.vscode/extensions/github.codespaces-1.10.0/src/resolver/tunnelFactory/handlePortSharingSucceededEvent.ts:38:17)
```

**Comment by matt-sre** [comment]
Trying to forward the port using `gh`:
```
gh cs ports forward 3000:3000 -c <Codespace name>
Forwarding ports: remote 3000 <=> local 3000
failed to share remote port 3000: error while starting port sharing: Basis tunnel and tunnel ports must be initialized
```

**Reply by peerkleio** [reply]
Popup with the same error in VS Code
<img width="477" alt="Screen Shot 2022-08-03 at 4 05 31 PM" src="https://user-images.githubusercontent.com/52968121/182712135-2184e8fa-a5b1-4b4a-8d95-b06e7b9bf6a5.png">

**Comment by sayhiben** [comment]
My team is also running into this error with `gh cs ssh`. The `--debug` flag doesn't reveal anything interesting - the log is empty.

```
kex_exchange_identification: read: Connection reset by peer
Connection reset by 127.0.0.1 port 55211
tunnel closed: failed to share remote port 2222: error while starting port sharing: Basis tunnel and tunnel ports must be initialized
```

**Comment by KartikM123** [comment]
Hi @peerkleio! Thanks for filing this bug!

We are currently deploying a hotfix for this bug. We've found that this is mainly affecting codespaces created in the US West region. As a temporary workaround, we suggest creating a codespace in any other region. 

To do this on github.com, you will need to navigate `github.com/codespaces/new`. On this page, create your codespace in any region other than  US West (see screenshot); if you are located in Western US I would recommend creating an US East codespace for best performance:
<img width="852" alt="image" src="https://user-images.githubusercontent.com/19193984/182721389-60ecb919-a214-4eae-9ee1-03e99b9ca283.png">

Thanks again for reporting the bug and let me know if you still are having issues!

**Comment by peerkleio** [comment]
We don't appear to be having the issue anymore. Thanks!

**Comment by benny-png** [comment]
Just use port-forwarding services like NGROK lol SUPER easy three line of shell commands https://ngrok.com/download
you can use this for reference https://youtu.be/AhiXVaWst24
